how-to-docs/certificates.md

```diff
@@ -0,0 +1,124 @@
+### **Application Certificates**
+-----
+
+The platform provide an in-house certificate authority for provide local hostname.namespace.svc.cluster.local certificates. Note: the CA is setup only to serve certificates under hostname.something.svc.clutser.local, ip addresses and single named hostnames, i.e. localhost, myservice etc.
+
+#### **CA Bundle**
+
+By default in the all namespaces a ca bundle has been added which can been mounted into the /etc/ssl/certs of the container and contains the root CA used to verify authenticity of the certificates. An example of using it is given below.
+
+#### **Example Deployment**
+
+Although technically the certificates aren't exclusively for ingress, the following show's a common deployment of external service via [ingress](https://github.com/UKHomeOffice/application-container-platform/blob/master/how-to-docs/ingress.md) with backend services / pods using tls between ingress controller and themselves.
+
+First we have to acquire the certficate from the CA.
+
+```
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: example
+spec:
+  replicas: 1
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 1
+  template:
+    metadata:
+      labels:
+        name: example
+    spec:
+      containers:
+      - name: certs
+        image: quay.io/ukhomeofficedigital/cfssl-sidekick:v0.0.1
+        imagePullPolicy: Always
+        args:
+        - --certs=/certs
+        - --domain=myservice.${KUBE_NAMESPACE}.svc.cluster.local
+        - --domain=another_domain_name
+        - --expiry=8760h
+        env:
+        - name: KUBE_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        volumeMounts:
+        # a emptyDir which the sidekicks writes the certificates
+        - name: certs
+          mountPath: /certs
+        # The platform CA Bundle hold the root ca used to verify the certificate chain
+        - name: bundle
+          mountPath: /etc/ssl/certs
+          readOnly: true
+      - name: you_application
+        image: quay.io/ukhomeofficedigital/some_image
+        ...
+        ports:
+        - name: https
+          port: 443
+          targetPort: 443
+        volumeMounts:
+        # You can configure you application to pick up the certificates from here (tls.pem and tls-key.pem)
+        - name: certs
+          mountPath: /certs
+          readOnly: true
+      # Ensure you mount the volumes
+      volumes:
+      - name: bundle
+        configMap:
+          name: bundle
+      - name: certs
+        emptyDir: {}
+```
+
+The cfssl sidekick with request the certificates and place into the /certs directory which is shared between app and itself. Note, the ```--expiry`` can be used to configure the rotation of the certificatesl, though it's presentky up the application to reload them.
```

Comment:
extra ` need after expiry. and presentky -> presently, and remove the extra l on certificates